### PR TITLE
Remove not used fields.

### PR DESCRIPTION
**Pull request by bernatnan**
Remove some fields not used.
Remove from invoice 2 SII fields not used any more in there. Now us the tax deffined.
Fix some bugs.

### CODE REVIEW

File: aeat.py [hunk 1]
```diff
@@ -942,10 +920,26 @@ class SIIReportLine(ModelSQL, ModelView):
...
+        table = cls.__table_handler__(module_name)
+        sql_table = cls.__table__()
+
+        exist_sii_excemption_key = table.column_exist('exemption_key')
+
+        super(SIIReportLine, cls).__register__(module_name)
+
+        if exist_sii_excemption_key:
+            # Don't use UPDATE FROM because SQLite nor MySQL support it.
+            cursor.execute(*sql_table.update([sql_table.exemption_cause],
```
Comment (albertca) [inline aeat.py]: Better use rename. Ie: https://hg.tryton.org/modules/sale_opportunity/file/tip/opportunity.py#l138

File: aeat_mapping.py [hunk 2]
```diff
@@ -31,8 +32,20 @@ class BaseInvoiceMapper(Model):
...
+
+    def exempt_kind(self, tax):
+        return attrgetter('sii_exemption_cause')(tax)
+
+    def not_subject(self, invoice):
+        base = 0
+        for line in invoice.lines:
+            for tax in line.taxes:
+                if tax.sii_exemption_cause == 'NotSubject':
+                    base += attrgetter('amount')(line)
```
Comment (albertca) [inline aeat_mapping.py:47]: It seems to me that there should be a "break" after "base += ....". After all, we don't want to sum the amount twice if there are two taxes with "NotSubject".

Though I understand this should never happen...

File: aeat_mapping.py [hunk 3]
```diff
@@ -233,6 +247,31 @@ def _build_issuer_id(self, invoice):
...
+            res['CuotaRecargoEquivalencia'] = (
+                self.tax_equivalence_surcharge_amount(tax))
+        return res
+
+    def location_rules(self, invoice):
+        base = 0
+        for line in invoice.lines:
+            for tax in line.taxes:
+                if tax.sii_issued_key == '08':
+                    base += attrgetter('amount')(line)
```
Comment (albertca) [inline aeat_mapping.py:272]: It seems to me that there should be a "break" after "base += ....". After all, we don't want to sum the amount twice if there are two taxes with "08".

Though I understand this should never happen...

File: aeat.py (hunk 1 above)
Comment (bernatnan) [reply aeat.py]: Done. I didn't know the rename function.

File: aeat_mapping.py (hunk 2 above)
Comment (bernatnan) [reply aeat_mapping.py:47]: > It seems to me that there should be a "break" after "base += ....". After all, we don't want to sum the amount twice if there are two taxes with "NotSubject".
> 
> Though I understand this should never happen...

It should, but if there are more than one tax defined as "NotSubject" we need to sum together.

The wrong is that the function is called to much times. I move out of the taxes for.

File: aeat_mapping.py (hunk 3 above)
Comment (bernatnan) [reply aeat_mapping.py:272]: > It seems to me that there should be a "break" after "base += ....". After all, we don't want to sum the amount twice if there are two taxes with "08".
> 
> Though I understand this should never happen...

The same as in the not_subject function. We need to ensure that there aren't more taxes defined identically.

**Pull request merged**